src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -15,12 +15,30 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal abstract class DecisionTree
     {
+        /// <summary>
+        /// The input expression to this branch of the decision tree.
+        /// </summary>
         public readonly BoundExpression Expression;
+        /// <summary>
+        /// The type of the input at this branch of the decision tree.
+        /// </summary>
         public readonly TypeSymbol Type;
+        /// <summary>
+        /// A temporary variable that is holding the computed input at this branch.
+        /// </summary>
         public LocalSymbol Temp;
+        /// <summary>
+        /// True if this decision tree fully handles all possible values of its input.
+        /// </summary>
         public bool MatchIsComplete;
 
+        /// <summary>
+        /// The three different kinds of nodes in the decision tree.
+        /// </summary>
         public enum DecisionKind { ByType, ByValue, Guarded }
```

Comment:
I prefer not to repeat the comments that are on the decision tree classes themselves. Are you concerned that this undermines the tests that are added in this PR?